router/usecase/pools/pool_factory.go

```
@@ -161,3 +145,66 @@ func newRoutableCosmWasmPool(pool sqsdomain.PoolI, cosmWasmConfig domain.CosmWas
+func newRoutableCosmWasmPoolWithCustomModel(
+	pool sqsdomain.PoolI,
+	cosmwasmPool *cwpoolmodel.CosmWasmPool,
+	cosmWasmConfig domain.CosmWasmPoolRouterConfig,
+	tokenOutDenom string,
+	takerFee osmomath.Dec,
+) (sqsdomain.RoutablePool, error) {
+	sqsPoolModel := pool.GetSQSPoolModel()
+
+	// Check if the pool is a transmuter pool with alloyed assets
+	model := sqsPoolModel.CosmWasmPoolModel
+	balances := sqsPoolModel.Balances
+	spreadFactor := sqsPoolModel.SpreadFactor
+	if model != nil {
+		// since v2, we introduce concept of alloyed assets but not yet actively used
+		// since v3, we introduce concept of normalization factor
+		// `routableAlloyTransmuterPoolImpl` is v3 compatible
+		_, isAlloyedTransmuterCodeId := cosmWasmConfig.AlloyedTransmuterCodeIDs[cosmwasmPool.CodeId]
+		if isAlloyedTransmuterCodeId && model.IsAlloyTransmuter() {
+			if model.Data.AlloyTransmuter == nil {
+				return nil, domain.CwPoolDataMissingError{
+					CosmWasmPoolType: domain.CosmWasmPoolAlloyTransmuter,
+					PoolId:           pool.GetId(),
+				}
+			}
+
+			return &routableAlloyTransmuterPoolImpl{
+				ChainPool:           cosmwasmPool,
+				AlloyTransmuterData: model.Data.AlloyTransmuter,
+				Balances:            balances,
+				TokenOutDenom:       tokenOutDenom,
+				TakerFee:            takerFee,
+				SpreadFactor:        spreadFactor,
+			}, nil
+		}
+
+		_, isOrderbookCodeId := cosmWasmConfig.OrderbookCodeIDs[cosmwasmPool.CodeId]
+		if isOrderbookCodeId && model.IsOrderbook() {
+			if model.Data.Orderbook == nil {
+				return nil, domain.CwPoolDataMissingError{
+					CosmWasmPoolType: domain.CosmWasmPoolOrderbook,
+					PoolId:           pool.GetId(),
+				}
+			}
+
+			return &routableOrderbookPoolImpl{
+				ChainPool:     cosmwasmPool,
+				Balances:      balances,
+				TokenOutDenom: tokenOutDenom,
+				TakerFee:      takerFee,
+				SpreadFactor:  spreadFactor,
+				OrderbookData: model.Data.Orderbook,
+			}, nil
+		}
+	}
+
+	return nil, nil
```

Could we return an error in this case? 


Returning nil, nil is an anti-pattern that encourages coupling this function to the context of its caller.